core/container/src/main/java/org/wildfly/swarm/container/config/ConfigViewFactory.java

```diff
@@ -166,14 +172,39 @@ private void loadYamlProjectConfig(String name, URL url) throws IOException {
 
     @SuppressWarnings("unchecked")
     private void loadYamlProjectConfig(String name, InputStream inputStream) {
-        Yaml yaml = new Yaml();
-        Map<String, ?> doc = (Map<String, ?>) yaml.load(inputStream);
+        Map<String, ?> doc = loadYaml(inputStream);
 
         ConfigNode node = MapConfigNodeFactory.load(doc);
         this.configView.register(name, node);
         this.configView.withProfile(name);
     }
 
+    static Map<String, ?> loadYaml(InputStream input) {
+        Yaml yaml = newYaml();
+        return (Map<String, ?>) yaml.load(input);
+    }
+
+    private static Yaml newYaml() {
+        return new Yaml(new Constructor(),
+                        new Representer(),
+                        new DumperOptions(),
+                        new Resolver() {
+                            @Override
+                            public Tag resolve(NodeId kind, String value, boolean implicit) {
+                                if (value != null) {
```

Comment:
We only want a partial special-case, not a complete retargetting of everything to STR.